### PR TITLE
Avoid repeated calls to stream finalizers

### LINKED ISSUE #937: Finalize handlers called twice on streams
A stream manager that receives an error seems to call the finalize handler twice: once with the actual error and then again with default-constructed error.

### PR DESCRIPTION
**Pull request by Neverlord**
Fixes #937.

### CODE REVIEW

**Comment by Neverlord** [comment]
@josephnoir sorry for the noise in this PR, I accidentally pushed commits into the branch for other stuff I'm working on. This is done and ready for review as of d52371c. :)

File: libcaf_core/src/stream_manager.cpp [hunk 1]
```diff
@@ -116,15 +113,21 @@ void stream_manager::handle(stream_slots slots, upstream_msg::ack_batch& x) {
...
+  CAF_LOG_TRACE(CAF_ARG(slots) << CAF_ARG(x));
   if (out().remove_path(slots.receiver, x.reason, true))
     stop(std::move(x.reason));
 }
 
 void stream_manager::stop(error reason) {
+  CAF_LOG_TRACE(CAF_ARG(reason));
+  if (stopped())
+    return;
+  flags_ = is_stopped_flag;
```
Comment (josephnoir) [inline libcaf_core/src/stream_manager.cpp]: `flags_` seems to store bit flags. I'm assuming overwriting them is intended here?

File: libcaf_core/src/stream_manager.cpp [hunk 2]
```diff
@@ -135,12 +138,12 @@ void stream_manager::stop(error reason) {
 
 void stream_manager::shutdown() {
   CAF_LOG_TRACE("");
-  // Mark as shutting down and reset other flags.
-  if (shutting_down())
```
Comment (josephnoir) [inline libcaf_core/src/stream_manager.cpp]: The header has a call to [`shutting_down()`](https://github.com/actor-framework/actor-framework/blob/d52371cff540c11d968b960a7643633371371902/libcaf_core/caf/stream_manager.hpp#L170) as well. I was wondering if that should test for `is_stopped_flag`. And if so, maybe `shutting_down()` should just check both flags?

File: libcaf_core/caf/stream_manager.hpp [hunk 3]
```diff
@@ -51,6 +51,10 @@ class stream_manager : public ref_counted {
   /// buffered elements.
   static constexpr int is_shutting_down_flag = 0x0002;
 
+  /// Denotes whether the manager has already stopped. Calling member functions
```
Comment (josephnoir) [inline libcaf_core/caf/stream_manager.hpp]:
```suggestion
  /// Denotes whether the manager has stopped. Calling member functions
```

File: libcaf_core/caf/stream_manager.hpp [hunk 4]
```diff
@@ -148,16 +152,21 @@ class stream_manager : public ref_counted {
     return getf(is_shutting_down_flag);
   }
 
+  /// Returns whether this manager has already stopped.
```
Comment (josephnoir) [inline libcaf_core/caf/stream_manager.hpp]:
```suggestion
  /// Returns whether this manager has stopped.
```

File: libcaf_test/caf/test/dsl.hpp [hunk 5]
```diff
@@ -632,12 +632,12 @@ class test_coordinator_fixture {
   // -- constructors, destructors, and assignment operators --------------------
 
   static Config& init_config(Config& cfg) {
+    cfg.set("logger.file-verbosity", caf::atom("quiet"));
```
Comment (josephnoir) [inline libcaf_test/caf/test/dsl.hpp]: Fixing #959 as a side effect here?

File: libcaf_core/src/stream_manager.cpp (hunk 1 above)
Comment (Neverlord) [reply libcaf_core/src/stream_manager.cpp]: Yes, it is. Setting stop clears all other flags.

File: libcaf_test/caf/test/dsl.hpp (hunk 5 above)
Comment (Neverlord) [reply libcaf_test/caf/test/dsl.hpp]: Apparently yes, I needed a way to look at the log output while working on this.

File: libcaf_core/src/stream_manager.cpp (hunk 2 above)
Comment (Neverlord) [reply libcaf_core/src/stream_manager.cpp]: Good catch. Will fix.

**Pull request merged**